DevTools/UserspaceEmulator/Emulator.cpp

```diff
@@ -125,54 +165,70 @@ void Emulator::setup_stack(const Vector<String>& arguments, const Vector<String>
 
 bool Emulator::load_elf()
 {
-    m_elf->image().for_each_program_header([&](const ELF::Image::ProgramHeader& program_header) {
+    MappedFile mapped_executable(m_executable_path);
+    if (!mapped_executable.is_valid()) {
+        reportln("Unable to map {}", m_executable_path);
+        return false;
+    }
+
+    ELF::Image executable_elf((const u8*)mapped_executable.data(), mapped_executable.size());
+
+    if (!executable_elf.is_dynamic()) {
+        // FIXME: Support static objects
+        ASSERT_NOT_REACHED();
+    }
+
+    String interpreter_path;
+    executable_elf.for_each_program_header([&interpreter_path](const ELF::Image::ProgramHeader& program_header) {
```

Comment:
I think ELF::validate_program_headers will do this for you, and do some extra checking along the way as well.

Reply:
That's nice, thanks!